numpy/testing/_private/utils.py

```diff
@@ -695,9 +695,14 @@ def isnumber(x):
     def istime(x):
         return x.dtype.char in "Mm"
 
-    def chk_same_position(x_id, y_id, hasval='nan'):
-        """Handling nan/inf: check that x and y have the nan/inf at the same
-        locations."""
+    def func_assert_same_pos(x, y, func=isnan, hasval='nan'):
+        """Handling nan/inf: combine results of running func on x and y,
+        checking that they are True at the same locations."""
+        x_id = func(x)
+        y_id = func(y)
+        if not any(x_id) and not any(y_id):
+            return False
```

Comment:
Are these two lines needed? Aren't they covered by the below?

Reply:
You'd think not, but numerous recursion failures if I leave them out...

Reply:
So, what happens is that it does recurse, but the `if` statement prevents it from doing it more than once.

Reply:
which is simple to solve - just use `any` - will push fix up shortly.

Reply:
Doesn't work either, but passing in `equal_[nan|inf]=False` does.

Reply:
No, it doesn't...

Reply:
Maybe add a comment saying that this is needed to prevent recursion or something?

Good to go either way